bootstore/src/db/error.rs

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! DB related errors
+
+use serde::{Deserialize, Serialize};
+
+#[derive(thiserror::Error, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum Error {
+    #[error("Failed to open db connection to {path}: {err}")]
+    DbOpen { path: String, err: String },
+
+    #[error("Diesel/sqlite error: {err}")]
+    Diesel { err: String },
+
+    #[error("BCS serialization error: {err}")]
+    Bcs { err: String },
+
+    // Temporary until the using code is written
+    #[allow(dead_code)]
+    #[error("Share commit for {epoch} does not match prepare")]
+    CommitHashMismatch { epoch: i32 },
+
+    #[error("No shares have been committed")]
+    NoSharesCommitted,
+
+    #[error("DB invariant violated: {0}")]
+    DbInvariant(String),
+
+    #[error("Already initialized with rack uuid: {0}")]
+    AlreadyInitialized(String),
+
+    #[error(
+        "Tried to Prepare a KeyShare with epoch {epoch}, but found one 
+    with later epoch {stored_epoch}"
```

Comment:
Formatting of this message is a little suspect; probably shouldn't be a newline in there?

Reply:
fixed.